content/ADR-204-comms-architecture.md

```diff
@@ -0,0 +1,620 @@
+  actor A as Alice
+  participant C as Catalyst
+  loop 
+    A->>A: Profile version is changed?
+    A->>B: Broadcast ProfileVersion($version)
+    B->>B: Do I have Alice's profile version==$version?
+    alt is Yes
+        B->>B: Do nothing
+    else is No    
+        B->>C: Request a new profile from Catalyst
+        C->>B: Profile Response
+        B->>B: Cache the new Alice Profile
+    end    
+  end
+```
+
+```protobuf
+message AnnounceProfileVersion {
+  uint32 profile_version = 1;
+}
+```
+
+`AnnounceProfileVersion` is sent to both `Island` and `Scene` rooms.
+The profile version should match the version sent to the catalyst.
+
+The process on the client looks like following:
+
+```mermaid
+flowchart LR
+id1["Make Profile Changes"] --> id2["Increment Version"] --> id3["Publish to Catalyst"] --> id4["Announce Profile Version"]
+```
+
+**Metadata**
+
+To ensure that peers will be able to retrieve the updated profile version from the correct Catalyst, `lambdasEndpoint` should be set in `Metadata`.
+It is equal to the `lambdas.publicUrl` received from `/about` response.
+It is sufficient to set this data only once per user session as it does not change while the session is active.
+
+`lambdasEndpoint` is optional and can be omitted: in this case it's not guaranteed that the profile received on a peer end will reflect the most recent changes, as it may take time to propagate profile changes through all available Catalysts.
+
+```
+{
+  ...
+  "lambdasEndpoint": <serverAbout.lambdas.publicUrl>,
+  ...
+}
+```
+
+### Player Emote
+
+```protobuf
+message PlayerEmote {
+  uint32 incremental_id = 1;
+  string urn = 2;
+}
+```
+
+`PlayerEmote` is sent to both `Island` and `Scene` rooms.
+
+This message is used to notify other peers that a user is performing an emote. The `incremental_id` is a monotonic counter that grows with every sent message.
+- `incremental_id`: A monotonic counter that grows with every sent message. It is used for messages deduplication 
+- `urn`: The Uniform Resource Name (URN) of the emote being performed.
+
+**Looping Emotes Replication**
+
+For a looping emote this message should be replicated every cycle to ensure its visibility for a newly joined peer.
+
+```mermaid
+sequenceDiagram
+  actor A as Peer
+  participant B as LiveKit  
+  A->>A: Play Emote "Dance"
+  A->>A: Increment ID
+  A->>B: Broadcast PlayerEmote(incremental_id, "Dance")
+  A->>A: Is the emote looping?
+  alt is Yes
+    loop every emote cycle
+        A->>A: Don't increment ID
+        A->>B: Broadcast PlayerEmote(incremental_id, "Dance")
+    end
+  end    
+  
+```
+
+> Note: when a peer joins a room they will miss information about emotes currently being played. If an emote is looping they will receive it with the next message
+
+### Movement (Compressed)
+
+```protobuf
+message MovementCompressed {
+  int32 temporal_data = 1; // bit-compressed: timestamp + animations 
+  int64 movement_data = 2; // bit-compressed: position + velocity 
+}
+```
```

Comment:
Hey folks, since adding all the compressed complexity (and the assumption in each part) is not priceless, I imagine you have an analysis report to share the trade-off here. Adding that analysis to understand the real motivation behind this would be interesting. Also, it'd be interesting to have the comparison in the practice (maybe benchmarks or theoretical scenarios):

- We have a 48-byte message (uncompressed) vs. a 12-byte message (`MovementCompreesed`). You could reduce the 48-byte message to 40-byte or fewer with significantly less commitment and assumptions. But let's take 48/12=4; this is the gain ratio. Have you considered the LiveKit overhead that this is sent on top of that? This is directly affected, and it results in a lower gain ratio. Then you also have the WebRTC overhead and then the UDP/IP layer (I imagine this is sent by the unreliable way)

- What are the rationale behind each constant? I can guess some of them, but let's not leave it to the imagination.

- Instead of having the constant, would it be more suitable to have a `MovementHeader` (to be announced with the profile maybe) where all the constant are set it up for that profile? It'd be far flexible and you avoid explain the constant here, only the limitations. For instance, you send the minValues, maxValues, nBits, etc.